src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/IDictionaryExtensions.cs

```diff
@@ -224,4 +224,42 @@ public static void MultiRemove<TKey, TValue>(this IDictionary<TKey, ImmutableArr
             }
         }
     }
+
+    /// <summary>
+    /// Removes entries from a dictionary based on a specified condition. The condition is defined by a function that
+    /// evaluates each key-value pair.
+    /// </summary>
+    public static void Remove<TKey, TValue, TArg>(this IDictionary<TKey, TValue> dictionary, Func<TKey, TValue, TArg, bool> predicate, TArg arg)
+        where TKey : notnull
+    {
+#if NET
+        // .NET supports removing while enumerating:
```

Comment:
is this documented/guaranteed?

Reply:
also, you're operating on IDictionary, not Dictionary here.  so this is assume *all* IDictionaries support this.

Reply:
ok.  he linked to https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.remove?view=net-9.0 which says:

> NET Core 3.0+ only: this mutating method may be safely called without invalidating active enumerators on the [Dictionary<TKey,TValue>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-9.0) instance. This does not imply thread safety.

So i'm ok with this *if* we make this an extension on Dict, not IDic.